src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,63 @@
+import { FC, ReactNode } from 'react'
+
+import { DialogContainerProps, DialogFooterItems } from './types'
+import { Dialog } from './dialog.atoms'
+import { useDialogAppearance } from './use-dialog-appearance'
+
```

Comment:
> I think there's been a misunderstanding of my previous comments on the interface that we should be aiming for

😅now I got the things that you want to deliver to me by looking this comment, the first pattern is an opinionated and that should be exists as Dialog molecule and that molecule just need to rely on some props without having a side effect like `useDialogAppearance` does

Reply:
I'm not sure I understand your comment above, so to make sure we're on the same page, I think the Dialog component export from this file just needs to do:

```javascript
// Omit `open` attribute because we don't want that available to consumers
// of our React-based Dialog component.
interface DialogAttributes
  extends Omit<DialogHTMLAttributes<HTMLDialogElement>, "open"> {}

export interface DialogProps extends DialogAttributes {
  children: ReactNode
  isOpen?: boolean
  onClose: () => void
  size?: "small" | "medium"
}

type DialogFC = FC<DialogProps> & {
  Body: typeof ElDialogBody
  Footer: typeof ElDialogFooter
  Header: typeof ElDialogHeader
  Title: typeof ElDialogTitle
}

export const Dialog: DialogFC = ({
  children,
  isOpen = false,
  onClose,
  size = "small",
}) => {
  const ref = useRef<HTMLDialogElement>(null)

  useToggleDialogVisibilityEffect({ isOpen, onClose, ref })

  return (
    <ElDialog data-size={size} ref={ref}>
      {children}
    </ElDialog>
  )
}

Dialog.Body = ElDialogBody
Dialog.Footer = ElDialogFooter
Dialog.Header = ElDialogHeader
Dialog.Title = ElDialogTitle
```

(Or something close to that)